File: ui/cms/src/model/movies/Movie.ts
export type MediaType = "POSTER" | "TRAILER" | "BACKDROP";
export enum MovieStatus {
  DRAFT = 0,
  PUBLISHED = 1,
  HIDDEN = 2,
}

export interface Genre {
  id: number;
  name: string;
}

export interface Country {
  name: string;
  code: string;
}

export interface Language {
  code: string;
  name: string;
}

export interface Media {
  id: number;
  type: MediaType;
  url: string;
}

interface Movie {
  id: string; // UUID
  originalTitle: string;
  localizedTitle: string;
  description: string;
  director: string;
  mainActors: string;
  duration: number; // en minutos
  originalReleaseYear: number;
  genres: Genre[];
  country: Country;
  originalLanguage: Language;
  trailerUrl: string;
  posterUrl: string;
  extraImageUrls: string[];
  status: MovieStatus;
  createdAt: Date;
  deletedAt: Date | null;
}

export interface MovieSearchFilters {
  searchTerm?: string;
  includeDeleted?: boolean;
  status?: MovieStatus[];
  page: number;
  pageSize: number;
}

export default Movie;

File: server/MoviesService/Models/Movie.cs
using Common.Enums;
using MoviesService.Enums;

namespace MoviesService.Models
{
    public class Movie
    {
        public Guid Id { get; set; }
        public required string OriginalTitle { get; set; }
        public required string LocalizedTitle { get; set; }
        public required string Description { get; set; }
        public required string Director { get; set; }
        public required string MainActors { get; set; }

        public required int Duration { get; set; }
        public required int OriginalReleaseYear { get; set; }
        public required string CountryCode { get; set; }
        public required string OriginalLanguageCode { get; set; }

        public required ICollection<MovieGenre> Genres{ get; set; } = new List<MovieGenre>();
        public required ICollection<MovieMedia> Media { get; set; } = new List<MovieMedia>();
        public required EntityStatus Status { get; set; } = EntityStatus.Draft;
        public required DateTime CreatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}
